### PR TITLE
ADM-812: [Frontend] prevent get board info when board config form field is dirty on metric page

### PR DESCRIPTION
**Pull request by K-Chow**
## Summary

...

## Before

_Description_
when click previous button on metrics page, then forward to metrics page, board config data will be refresh.

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After
if user input or change any from field about board config, do not refresh data when go back，then reenter the metrics page

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

File: frontend/src/containers/MetricsStepper/index.tsx [hunk 1]
```diff
@@ -261,7 +260,7 @@ const MetricsStepper = () => {
         branches: [],
       }));
       dispatch(updatePipelineList(initPipelineSteps));
-      dispatch(updateCycleTimeSettings([]));
+      // dispatch(updateCycleTimeSettings([]));
```
Comment (guzhongren) [inline frontend/src/containers/MetricsStepper/index.tsx:263]: ？？？

File: frontend/src/context/Metrics/metricsSlice.ts [hunk 2]
```diff
@@ -34,6 +34,7 @@ export interface ICycleTimeSetting {
 }
 
 export interface savedMetricsSettingState {
+  isDirty: boolean;
```
Comment (guzhongren) [inline frontend/src/context/Metrics/metricsSlice.ts:37]: isDirty what?

**Review by guzhongren — COMMENTED** [review]
change with the comments

**Review by SimonTal — APPROVED** [review]
Approved

**Pull request merged**